builtin/providers/librato/resource_librato_alert.go

```diff
@@ -0,0 +1,446 @@
+package librato
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"math"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/henrikhodne/go-librato/librato"
+)
+
+func resourceLibratoAlert() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLibratoAlertCreate,
+		Read:   resourceLibratoAlertRead,
+		Update: resourceLibratoAlertUpdate,
+		Delete: resourceLibratoAlertDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+			"id": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"active": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"rearm_seconds": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  600,
+			},
+			"services": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"condition": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"metric_name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"source": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"detect_reset": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"duration": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"threshold": &schema.Schema{
+							Type:     schema.TypeFloat,
+							Optional: true,
+						},
+						"summary_function": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+				Set: resourceLibratoAlertConditionsHash,
+			},
+			"attributes": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"runbook_url": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			// TODO add missing condition attrs
+		},
+	}
+}
+
+func resourceLibratoAlertConditionsHash(v interface{}) int {
```

Comment:
I didn't add any more because in practice I didn't really consider using multiple layers of alerting on the same metric, but I guess you're right and it's better to be safe than sorry. I'll add those in.